plutus-pab/src/Cardano/Protocol/Socket/Server.hs

```diff
@@ -108,6 +108,23 @@ trimTo :: MonadIO m => ServerHandler -> Int -> m ()
 trimTo ServerHandler {shCommandChannel} size =
     liftIO $ atomically $ writeTQueue (ccCommand shCommandChannel) $ TrimTo size
 
+{- Create a thread that keeps the number of blocks in the channel to the maximum
+   limit of K -}
+pruneChain :: MonadIO m => Integer -> TChan Block -> m ThreadId
+pruneChain k original = do
+  localChannel <- liftIO $ atomically $ cloneTChan original
+  liftIO . forkIO $ go k localChannel
+  where
+  go :: MonadIO m => Integer -> TChan Block -> m ()
+  go k' localChannel = do
+    -- Wait for data on the channel
+    _ <- liftIO $ atomically $ readTChan localChannel
+    if k' == 0
+       -- Remove the blocks from the original channel. New connections will not
+       -- receive these blocks.
+       then liftIO $ atomically (readTChan original) >> go 0 localChannel
```

Comment:
So 
* we start with `k'=k`  and `localChannel` set to a clone of `original`
* We read `k` blocks, decreasing `k'` each time
* When we get to `k' == 0`, we read everything from `original` and start again with `k' = 0`

Every `k` blocks we delete `k` blocks. But I think what it should do is: Every block, delete the oldest block if there are more than `k` blocks. So that the number of blocks goes up until it reaches `k` and then stays at `k`.

Reply:
Or am I missing anything?

Reply:
I think `readTChan` reads only one Block, so it should do what you are suggesting.

Reply:
Ah yes. Thanks.

Reply:
I see how the comment can be misleading. I will change it.